server/middlewares/authValidation.js

```diff
@@ -0,0 +1,24 @@
+import validate from './validate';
+/** User authentication validation */
+
+/**
+   * validate user sign up endpoint
+   * @param {Object} req - request body
+   * @param {Object} res - response body
+   * @param {Object} next - call next function
+   * @returns {Function} validate - call validate function
+   */
+
+export const validateUserSignUp = (req, res, next) => {
+  const data = req.body;
+  const signUpRules = {
+    email: 'required|email',
+    username: 'required|string|alpha',
+    firstName: 'required|string|alpha',
```

Comment:
I don't think it is a good idea to use the alpha rule for the first name and last name, you may want to consider using string rule.
Also, I'd suggest you give a minimum value for first name, last name, username to avoid scenarios where the user inputs just one character.